apis/vlan/v1alpha1/virtualnetwork_types.go

```diff
@@ -0,0 +1,120 @@
+
+// VirtualNetworkSpec defines the desired state of VirtualNetwork
+type VirtualNetworkSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  VirtualNetworkParameters `json:"forProvider"`
+}
+
+// VirtualNetworkStatus defines the observed state of VirtualNetwork
+type VirtualNetworkStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     VirtualNetworkObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// VirtualNetwork is a managed resource that represents a Packet VirtualNetwork
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.atProvider.state"
+// +kubebuilder:printcolumn:name="ID",type="string",JSONPath=".status.atProvider.id"
+// +kubebuilder:printcolumn:name="HOSTNAME",type="string",JSONPath=".spec.forProvider.hostname"
+// +kubebuilder:printcolumn:name="FACILITY",type="string",JSONPath=".status.atProvider.facility"
+// +kubebuilder:printcolumn:name="IPV4",type="string",JSONPath=".status.atProvider.ipv4"
+// +kubebuilder:printcolumn:name="RECLAIM-POLICY",type="string",JSONPath=".spec.reclaimPolicy"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,packet}
+type VirtualNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VirtualNetworkSpec   `json:"spec,omitempty"`
+	Status VirtualNetworkStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// VirtualNetworkList contains a list of VirtualNetworks
+type VirtualNetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VirtualNetwork `json:"items"`
+}
+
+// VirtualNetworkParameters define the desired state of a Packet Virtual Network.
+// https://www.packet.com/developers/api/vlans/#create-an-virtual-network
+//
+// Reference values are used for optional parameters to determine if
+// LateInitialization should update the parameter after creation.
+type VirtualNetworkParameters struct {
+	// +immutable
+	// +required
+	Facility string `json:"facility"`
+
+	// +optional
+	Description *string `json:"description,omitempty"`
+}
+
+// VirtualNetworkObservation is used to reflect in the Kubernetes API, the observed
+// state of the VirtualNetwork resource from the Packet API.
+type VirtualNetworkObservation struct {
+	ID           string       `json:"id"`
+	Href         string       `json:"href,omitempty"`
+	VXLAN        int          `json:"vxlan,omitempty"`
+	FacilityCode string       `json:"facility_code,omitempty"`
```

Comment:
I don't know if "Facility: any" is accepted for VLAN. If it is, this makes sense. Otherwise we could likely get away with only a spec facility.

Reply:
@displague do you know if `Facility` and `FacilityCode` are different values on packet side?

Reply:
Facility usually indicates a Facility an object in the Packet API.  Facility is also commonly used as  short-name for `Facility.Code`.

The shape of Facility is `{ID: "uuid", Name: "Somewhere in the world", Code: "EWR1"}`

This `facility` in the Device spec currently maps to the `Facility.Code`.


In the device

Reply:
Looking at https://www.packet.com/developers/api/vlans/#create-an-virtual-network, I think it is fair to call this field `facility`.
```
{
	"project_id": "string",
	"description": "string",
	"facility": "string",
	"vxlan": "integer",
	"vlan": "integer"
} 
```

Reply:
Err, I take that back. You are already calling it `facility` for the `spec`.  I defer to my previous comment https://github.com/packethost/crossplane-provider-packet/pull/18#discussion_r455193895

Reply:
Hmm if we don't know it seems like we may want to keep them both for now?